utilities/src/main/java/io/syndesis/qe/endpoints/publicendpoint/PublicEndpoint.java

```diff
@@ -0,0 +1,45 @@
+package io.syndesis.qe.endpoints.publicendpoint;
+
+import io.syndesis.qe.TestConfiguration;
+import io.syndesis.qe.templates.PublicOauthProxyTemplate;
+import io.syndesis.qe.utils.PublicApiUtils;
+import io.syndesis.qe.utils.RestUtils;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * Abstract class for public endpoints
+ */
+@Slf4j
+public abstract class PublicEndpoint {
+
+    protected String rootEndPoint = "/public";
+    private static final String apiPath = TestConfiguration.syndesisRestApiPath();
+    private static Client client;
+    private MultivaluedMap<String, Object> COMMON_HEADERS = new MultivaluedHashMap<>();
+
+    public PublicEndpoint(String endpoint) {
+        client = RestUtils.getClient();
+        COMMON_HEADERS.add("X-Forwarded-User", "pista");
+        COMMON_HEADERS.add("X-Forwarded-Access-Token", "kral");
+        COMMON_HEADERS.add("SYNDESIS-XSRF-TOKEN", "awesome");
+        COMMON_HEADERS.add("Authorization", "Bearer " + PublicApiUtils.getPublicToken());
+        rootEndPoint += endpoint;
+    }
+
+    String getWholeUrl(String publicEndpointUrl) {
+        return String.format("https://%s%s%s", PublicOauthProxyTemplate.PUBLIC_API_PROXY_ROUTE, apiPath, publicEndpointUrl);
+    }
+
+    Invocation.Builder createInvocation(String url) {
+        log.info(String.format("Creating invocation for url %s", url));
+        return client.target(url)
+                .request(MediaType.APPLICATION_JSON)
+                .headers(COMMON_HEADERS);
```

Comment:
I think this can be refactored into RestUtils directly so that we'll avoid duplicating the same code as in AbstractEndpoint. Something like the `client` would be in the `RestUtils` class and it will also have the `createInvocation` method.. but if you prefer we can log a jira for that and do it in separate PR

Reply:
Yes, it can be refactored in some way. However, the AbstractEndpoint and PublicEndpoint are differently (PublicEndpoint uses Authorization token, design of AbstractEndpoint cannot be applied on PublicApi, ) and now I don't know how to unify it without huge refactoring. And it should cover all class which uses Client.
e.g.
We can create some class (ClientManager ?) which will manage Client instance as common headers, creating an invocation with specific url ane specific media type etc. And this class (ClientManager ?) will use all classes (AbstractEndpoint, PublicEndpoint, ApiProviderSteps, CustomApiEndpoint, Verifier ...)

Reply:
something like that I had in my mind as well.. I'll create a jira for that

Reply:
ok, after REST tests I can look at it. In my POV, we can skip it for this PR.